tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -539,3 +541,25 @@ def _discover(obj):
         assert total_count == 10
     else:
         assert total_count == 9
+
+
+@cocotb.test(expect_error=ValueError)
+async def test_invalid_discovery_method(dut):
+    """Try accessing with an enum value for _GPIDiscovery out of bounds."""
+    dut._get("testsignal", discovery_method=5)
+
+
+@cocotb.test()
+async def test_none_return_on_invalid_signal(dut):
+    """Try accessing a signal that does not exist and make sure we get None back."""
+    assert dut._get("notexistingsignal") is None
+    assert dut._get("notexistingsignal", discovery_method=GPIDiscovery.AUTO) is None
+    assert dut._get("notexistingsignal", discovery_method=GPIDiscovery.NATIVE) is None
+
+
+@cocotb.test()
+async def test_native_discovery(dut):
+    """Try accessing a signal using native strategy."""
+    assert dut._get("stream_in_data") is not None
+    assert dut._get("stream_in_data", discovery_method=GPIDiscovery.AUTO) is not None
+    assert dut._get("stream_in_data", discovery_method=GPIDiscovery.NATIVE) is not None
```

Comment:
I'd like to see a test where NATIVE doesn't pick up an object, but AUTO does. Like on a mixed language boundary. That will probably have to be put into `test_iteration_mixedlang`.

Reply:
good point. will do.

Reply:
just commited, let's see how this goes.